system_test.go

```diff
@@ -0,0 +1,241 @@
+	if err != nil {
+		t.Fatalf("unexpected error in CreateBranchAndCheckout() %s", err)
+	}
+
+	// Create a commit and push
+	err = os.WriteFile(path.Join(workdir, "some-file.txt"), []byte("some-content"), 0644)
+	if err != nil {
+		t.Fatalf("unexpected error writing a file to git repo %s", err)
+	}
+	_, err = localRepository.AddAll()
+	if err != nil {
+		t.Fatalf("unexepected error in AddAll() %s", err)
+	}
+	err = localRepository.Commit("build: add test file")
+	if err != nil {
+		t.Fatalf("unexpected error in Commit() %s", err)
+	}
+	err = localRepository.Push(branchName)
+	if err != nil {
+		t.Fatalf("unexpected error in Push() %s", err)
+	}
+
+	// Create a pull request
+	client := github.NewClient(testToken, repo)
+	metadata, err := client.CreatePullRequest(t.Context(), repo, branchName, "main", "test: integration test", "do not merge")
+	if err != nil {
+		t.Fatalf("unexpected error in CreatePullRequest() %s", err)
+	}
+	t.Logf("created pull request: %d", metadata.Number)
+
+	// Add a label to the pull request
+	labels := []string{"do not merge", "type: cleanup"}
+	err = client.AddLabelsToIssue(t.Context(), repo, metadata.Number, labels)
+	if err != nil {
+		t.Fatalf("unexpected error in AddLabelsToIssue() %s", err)
+	}
+
+	// Get labels and verify
+	foundLabels, err := client.GetLabels(t.Context(), metadata.Number)
+	if err != nil {
+		t.Fatalf("unexpected error in GetLabels() %s", err)
+	}
+	if diff := cmp.Diff(foundLabels, labels); diff != "" {
+		t.Fatalf("GetLabels() mismatch (-want + got):\n%s", diff)
+	}
+
+	// Replace labels
+	labels = []string{"foo", "bar"}
+	err = client.ReplaceLabels(t.Context(), metadata.Number, labels)
+	if err != nil {
+		t.Fatalf("unexpected error in ReplaceLabels() %s", err)
+	}
+
+	// Get labels and verify
+	foundLabels, err = client.GetLabels(t.Context(), metadata.Number)
+	if err != nil {
+		t.Fatalf("unexpected error in GetLabels() %s", err)
+	}
+	if diff := cmp.Diff(foundLabels, labels); diff != "" {
+		t.Fatalf("GetLabels() mismatch (-want + got):\n%s", diff)
+	}
+
+	// Add label
+	err = client.AddLabelsToIssue(t.Context(), repo, metadata.Number, []string{"librarian-test", "asdf"})
+	if err != nil {
+		t.Fatalf("unexpected error in AddLabelsToIssue() %s", err)
+	}
+
+	// Get labels and verify
+	wantLabels := []string{"foo", "bar", "librarian-test", "asdf"}
+	foundLabels, err = client.GetLabels(t.Context(), metadata.Number)
+	if err != nil {
+		t.Fatalf("unexpected error in GetLabels() %s", err)
+	}
+	if diff := cmp.Diff(foundLabels, wantLabels); diff != "" {
+		t.Fatalf("GetLabels() mismatch (-want + got):\n%s", diff)
+	}
+
+	// Search for pull requests (this may take a bit of time, so try 3 times)
+	found := false
+	for i := 0; i < 3; i++ {
+		foundPullRequests, err := client.SearchPullRequests(t.Context(), "label:librarian-test is:open")
+		if err != nil {
+			t.Fatalf("unexpected error in SearchPullRequests() %s", err)
+		}
+		if len(foundPullRequests) == 0 {
```

Comment:
Why the length of `foundPullRequests` is 0?

Reply:
Oops that's backwards, will fix

Reply:
Done